packages/cli/src/executions/execution-recovery.service.ts

```diff
@@ -20,12 +21,15 @@ export class ExecutionRecoveryService {
 	constructor(
 		private readonly push: Push,
 		private readonly executionRepository: ExecutionRepository,
+		private readonly orchestrationService: OrchestrationService,
 	) {}
 
 	/**
 	 * Recover key properties of a truncated execution using event logs.
 	 */
 	async recoverFromLogs(executionId: string, messages: EventMessageTypes[]) {
+		if (this.orchestrationService.isFollower) return;
```

Comment:
Would it make sense to move this check to [this line](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts#L174) instead?

Reply:
My reasoning for making the leadership check internal to the service is to prevent someone in future using `recoverFromLogs` elsewhere and forgetting to account for leadership. This follows the pattern in `WaitTracker` and `PruningService`.